pkg/broker/handler/pool/options_test.go

```diff
@@ -61,25 +76,41 @@ func TestWithReceiveSettings(t *testing.T) {
 		NumGoroutines: 10,
 		MaxExtension:  time.Minute,
 	}
-	opt := NewOptions(WithPubsubReceiveSettings(want))
+	// Always add project id because in some env the default value can't be retrieved.
+	opt, err := NewOptions(WithPubsubReceiveSettings(want), WithProjectID("pid"))
+	if err != nil {
+		t.Errorf("NewOptions got unexpected error: %v", err)
+	}
 	if diff := cmp.Diff(want, opt.PubsubReceiveSettings); diff != "" {
 		t.Errorf("options ReceiveSettings (-want,+got): %v", diff)
 	}
 }
 
 func TestWithPubsubClient(t *testing.T) {
-	opt := NewOptions()
+	// Always add project id because in some env the default value can't be retrieved.
+	opt, err := NewOptions(WithProjectID("pid"))
+	if err != nil {
+		t.Errorf("NewOptions got unexpected error: %v", err)
+	}
 	if opt.PubsubClient != nil {
 		t.Errorf("options PubsubClient got=%v, want=nil", opt.PubsubClient)
 	}
-	opt = NewOptions(WithPubsubClient(&pubsub.Client{}))
+	// Always add project id because in some env the default value can't be retrieved.
```

Comment:
nit: in some env the default value can't be retrieved -> the default value can only be retrieved if running in a GKE cluster

Reply:
Apparently it also works on my cloudtop :) I think it's a GCE thing.

Reply:
yes you are right

Reply:
Done